tests/contrib/test_purl2url.py

```diff
@@ -134,7 +136,10 @@ def test_purl2url_get_inferred_urls():
             "https://gitlab.com/tg1999/firebase",
             "https://gitlab.com/tg1999/firebase/-/archive/1a122122/firebase-1a122122.tar.gz",
         ],
-        "pkg:pypi/sortedcontainers@2.4.0": ["https://pypi.org/project/sortedcontainers/2.4.0/"],
+        "pkg:pypi/sortedcontainers@2.4.0": [
+            "https://pypi.org/project/sortedcontainers/2.4.0/"
+        ],
+        "pkg:cocoapods/AFNetworking@4.0.1": ["https://cocoapods.org/pods/AFNetworking"],
```

Comment:
Add a test case where there is no name, IMO that will fail

Reply:
As noted in my other comments re removing the various try/except blocks I'd added, a test case with no name causes the entire test to fail, so I've removed the test case I'd previously added.

Reply:
I think this works as it is.